File: TUKulinarium_API/Data/Models/UserProfile.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TUKulinarium_API.Data.Models
{
    public class UserProfile
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid ProfileId { get; set; }
        [Required]
        [MaxLength(100)]
        public string UserName { get; set; }
        [Required]
        [MaxLength(50)]
        public string FirstName { get; set; }
        [Required]
        [MaxLength(50)]
        public string LastName { get; set; }
        [Required]
        [EmailAddress(ErrorMessage = "Please enter a valid email address")]
        public string Email { get; set; }
        [Required]
        [Phone(ErrorMessage = "Enter a valid number")]
        public string PhoneNumber { get; set; }
        public string Address { get; set; }

        [ForeignKey("User")]
        public string UserId { get; set; }
        public User User { get; set; }
        public ICollection<Order> Orders { get; set; }
        public ICollection<PaymentInfo> Payments { get; set; }
    }
}

File: TUKulinariumClient/src/app/models/UserProfile.ts
export class UserProfile {
   UserName: string;

   FirstName: string;

   LastName: string;

   Email: string;

   PhoneNumber: string;

   Address: string;
  
  constructor(
    username: string,
    firstName: string,
    lastName: string,
    email: string,
    phoneNumber: string,
    address: string
  ) {
    this.UserName = username;
    this.FirstName = firstName;
    this.LastName = lastName;
    this.Email = email;
    this.PhoneNumber = phoneNumber;
    this.Address = address;
  }
}
